kymatio/scattering2d/backend/backend_torch.py

```diff
@@ -18,22 +18,24 @@ def __init__(self, pad_size, pre_pad=False):
 
             Parameters
             ----------
-            pad_size : int
+            pad_size : list of 4 integers
                 size of padding to apply.
             pre_pad : boolean
                 if set to true, then there is no padding, one simply adds the imaginarty part.
         """
         self.pre_pad = pre_pad
-        self.padding_module = ReflectionPad2d(pad_size)
+        self.pad_size = pad_size
 
-    def __call__(self, input):
-        if(self.pre_pad):
-            output = input.new_zeros(input.size(0), input.size(1), input.size(2), input.size(3), 2)
-            output.narrow(output.ndimension()-1, 0, 1)[:] = input
-        else:
-            out_ = self.padding_module(input)
-            output = input.new_zeros(*(out_.size() + (2,)))
-            output.select(4, 0)[:] = out_
+        self.build()
+
+    def build(self):
+        self.padding_module = ReflectionPad2d(self.pad_size)
+
+    def __call__(self, x):
+        if not self.pre_pad:
+            x = self.padding_module(x)
+        output = x.new_zeros(x.shape + (2,))
```

Comment:
For the record, this could have been fixed with `torch.zeros(x.shape + (2,), dtype=x.dtype, device=x.device)`, but this is fine.

Reply:
I had that originally, but figured this was cleaner. After all, one of the bugs was caused by `torch.zeros(x.shape + (2,), dtype=x.dtype)`, where we didn't specify `device`. Before the PR, we also had `new_zeros`.